backend/assessments/1/src/main/java/org/example/IPLApplication2.java

```diff
@@ -0,0 +1,271 @@
+                ", average=" + average +
+                ", strikeRate=" + strikeRate +
+                ", wickets=" + wickets +
+                '}';
+    }
+}
+
+class Team {
+    private String teamName;
+    private List<Player> players;
+
+    public Team(String teamName) {
+        this.teamName = teamName;
+        this.players = new ArrayList<>();
+    }
+
+    public String getTeamName() {
+        return teamName;
+    }
+
+    public List<Player> getPlayers() {
+        return players;
+    }
+
+    public void addPlayer(Player player) {
+        players.add(player);
+    }
+}
+
+public class IPLApplication2 {
+    private static List<Team> teams = new ArrayList<>();
+
+    public static void main(String[] args) {
+        loadDataFromCSV("src/main/resources/IPL_2021-data.csv");
+
+        Scanner scanner = new Scanner(System.in);
+        int choice;
+
+        do {
+            Logging.logInfo("\nMenu:");
+            Logging.logInfo("1. Bowlers with at least 40 wickets for a given team");
+            Logging.logInfo("2. Details of the highest wicket-taker and highest run-scorer for a given team");
+            Logging.logInfo("3. Top 3 run-scorers and top 3 wicket-takers of the season");
+            Logging.logInfo("4. Generate match fixtures");
+            Logging.logInfo("5. Exit");
+            Logging.logInfo("Enter my choice: ");
+
+            choice = scanner.nextInt();
+            scanner.nextLine();
+
+            switch (choice) {
+                case 1:
+                    Logging.logInfo("Enter team name: ");
+                    String teamName = scanner.nextLine();
+                    Logging.logInfo("Enter minimum wickets: ");
+                    int minWickets = scanner.nextInt();
+                    List<Player> bowlers = findBowlersWithMinimumWickets(teamName, minWickets);
+                    Logging.logInfo("Bowlers with at least " + minWickets + " wickets for " + teamName + ":");
+                    for (Player bowler : bowlers) {
+                        Logging.logInfo(bowler.getName() + " - " + bowler.getWickets() + " wickets");
+                    }
+                    break;
+                case 2:
+                    Logging.logInfo("Enter team name: ");
+                    teamName = scanner.nextLine();
+                    displayHighestWicketTakerAndRunScorer(teamName);
+                    break;
+                case 3:
+                    displayTop3RunScorersAndWicketTakers();
+                    break;
+                case 4:
+                    generateMatchFixtures();
+                    break;
+                case 5:
+                    Logging.logInfo("Exiting the application. Goodbye!");
+                    break;
+                default:
+                    Logging.logInfo("Invalid choice. Please enter a valid option.");
+            }
+        } while (choice != 5);
+    }
+
+    private static void loadDataFromCSV(String filePath) {
```

Comment:
More modularity expected.
Move utils, core logic etc in seperate classes/packages